prometheus_client/registry.py

```diff
@@ -135,10 +135,23 @@ def __init__(self, names, registry):
         self._registry = registry
 
     def collect(self):
-        for metric in self._registry.collect():
-            m = metric._restricted_metric(self._name_set)
-            if m:
-                yield m
+        names = copy.copy(self._name_set)
+        collectors = set()
+        yield_target_info = False
```

Comment:
I'd have this as the thing to yield, rather than a bool for cleanliness. Also avoids a race.

Reply:
Done.